client/actions/user.js

```diff
@@ -0,0 +1,110 @@
+import axios from 'axios';
+
+/**
+ * @class UserActions
+ */
+export default class UserActions {
+  /**
+  * Request to the API to register a user
+  *
+  * @static
+  * @param {String} username The username of the user
+  * @param {String} email The email of the user
+  * @param {String} password The password of the user
+  * @returns {Object} dispatch object
+  * @memberof UserActions
+  */
+  static registerUser(username, email, password) {
+    return (dispatch) => {
+      dispatch({ type: 'REGISTRATION_BEGINS' });
+      return axios.post('https://postit-api.herokuapp.com/api/user/signup', {
+        username,
+        email,
+        password
+      })
+        .then((response) => {
+          dispatch({ type: 'REGISTRATION_SUCCESSFUL', payload: response.data });
+        })
+        .catch((err) => {
+          dispatch({ type: 'REGISTRATION_REJECTED', payload: err });
+        });
+    };
+  }
+
+  /**
+  * Request to the API to log a user in
+  *
+  * @static
+  * @param {String} username The username of the user
+  * @param {String} password The password of the user
+  * @returns {Object} dispatch object
+  * @memberof UserActions
+  */
+  static loginUser(username, password) {
+    return (dispatch) => {
+      dispatch({ type: 'LOGIN_BEGINS' });
+      return axios.post('https://postit-api.herokuapp.com/api/user/signin', {
+        username,
+        password
+      })
+        .then((response) => {
+          if (response.data.message === 'You are now logged in') {
```

Comment:
This is not necessary. If the request is successful then it will get here, if not it will return an error.